### PR TITLE
feat: allows seperate prefixTag version sequences

### PR DESCRIPTION
**Pull request by jahead**
Solves #572

### CODE REVIEW

**Comment by jvitor83** [comment]
@bcoe need that PR merged and a new release please.

**Comment by jahead** [comment]
there's conflicts I'll fix them on the weekend and update

**Comment by jvitor83** [comment]
@jahead don't forget that please. :)

**Comment by jahead** [comment]
Done. Sorry for taking so long. Let me know if anything else is needed

File: test/git.spec.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ function mock ({ bump, changelog, tags }) {
     }
   }))
 
-  mockery.registerMock('git-semver-tags', function (cb) {
+  mockery.registerMock('git-semver-tags', function (_, cb) {
```
Comment (bcoe) [inline test/git.spec.js:69]: @jahead this is looking good to me, but any chance we could add a test that covers the additional functionality?

Comment (jahead) [reply test/git.spec.js:69]: Yea mate, I"ll do it tonight. 

I was thinking how to test this

Comment (jahead) [reply test/git.spec.js:69]: @bcoe  Hey so I added some tests here and tested tagPrefix in general, as there didn't seem to be any tests already that could see. 

I'm not overly happy with is this as ` git-semver-tags` actually is the one filtering the tags, and due to the mock we won't know if  `git-semver-tags` changes will break the logic here. But the tests cover the use case in #702 

resolve if you're happy

**Comment by jahead** [comment]
Oddly, through testing this. 
I discovered/re-discovered that 
```
    if (pkg) {
      version = pkg.version
    } else if (args.gitTagFallback) {
      version = await latestSemverTag(args.tagPrefix)
    } else {
      throw new Error('no package file found')
    }
```

Which means that if there is a `packageFiles` defined we never get to use the gitSemverTags logic to resolve versions of different prefixes. And there's no option to force resolution by tag if `packageFiles` are defined. 

It will solve #702 but people should take care to configure no packageFiles. 

I'm not sure we should solve that problem in this PR, but it did seem like odd to me. 
I've outlined it in the tests.

**Review by bcoe — APPROVED** [review]

**Comment by bcoe** [comment]
@jahead released in `v9.2.0`, and appreciated. Let me know if there are any issues.

I do gradually try to make my way through @ messages on GitHub, so feel free to keep engaging in this way.

**Pull request merged**